auction/contracts/AuctionHouse.sol

```diff
@@ -0,0 +1,338 @@
+    function buyOutVerify(uint _auctionId, uint newAmount) internal view returns (bool) {
+        LibAucDataV1.DataV1 memory aucData = LibAucDataV1.parse(auctions[_auctionId].data, auctions[_auctionId].dataType);
+
+        if (aucData.buyOutPrice > 0 && aucData.buyOutPrice <= newAmount) {
+            return true;
+        }
+        return false;
+    }
+
+    /// @dev finishes, deletes and transfers all assets for an auction if it's ended (it exists, it has at least one bid, now > endTme)
+    function finishAuction(uint _auctionId) external {
+        require(checkAuctionExistence(_auctionId), "there is no auction with this id");
+        require(
+            !checkAuctionRangeTime(_auctionId) &&
+            auctions[_auctionId].buyer != address(0),
+            "only ended auction with bid can be finished");
+        doTransfers(_auctionId);
+        deactivateAuction(_auctionId);
+    }
+
+    /// @dev transfers auction assets (sellAsset to buyer, buyAsset to seller)
+    function doTransfers(uint _auctionId) internal {
+        Auction memory currentAuction = auctions[_auctionId];
+        address seller = currentAuction.seller;
+        address buyer = currentAuction.buyer;
+        if (buyer != address(0x0)) {//bid exists
+            LibOrderDataV2.DataV2 memory bidData = LibBidDataV1.getPaymentData(currentAuction.lastBid.data, currentAuction.lastBid.dataType);
+            if (bidData.payouts.length == 0){
+                LibPart.Part[] memory payout = new LibPart.Part[](1);
+                payout[0].account = payable(buyer);
+                payout[0].value = 10000;
+                bidData.payouts = payout;
+            }
+            LibOrderDataV2.DataV2 memory aucData = LibAucDataV1.getPaymentData(currentAuction.data, currentAuction.dataType);
+            if (aucData.payouts.length == 0){
+                LibPart.Part[] memory payout = new LibPart.Part[](1);
+                payout[0].account = payable(seller);
+                payout[0].value = 10000;
+                aucData.payouts = payout;
+            }
+            doTransfersWithFees(
+                currentAuction.lastBid.amount, 
+                address(this), 
+                MatchFees(currentAuction.protocolFee, currentAuction.protocolFee, LibFeeSide.FeeSide.TAKE),
+                bidData, 
+                aucData, 
+                currentAuction.buyAsset, 
+                currentAuction.sellAsset.assetType, 
+                TO_MAKER
+            );
+            transferPayouts(currentAuction.sellAsset.assetType, currentAuction.sellAsset.value, address(this), bidData.payouts, TO_TAKER);
+        } else {
+            transfer(currentAuction.sellAsset, address(this), seller, TO_SELLER, UNLOCK); //nft back to seller
+        }
+    }
+
+    /// @dev returns true if auction started and hasn't finished yet, false otherwise
+    function checkAuctionRangeTime(uint _auctionId) public view returns (bool){
+        uint endTime = auctions[_auctionId].endTime;
```

Comment:
if we initialize a variable right before its usage, we can decrease gas usage in case the check by aucData is failed. In this case, "endTime" will be unnecessary.

Reply:
fixed